nikola/post.py

```diff
@@ -1023,6 +1031,28 @@ def _get_metadata_from_file(meta_data):
     if not meta_data[0]:
         meta_data = meta_data[1:]
 
+    # If 1st line is '---', then it's YAML metadata
```

Comment:
Perhaps it should be more generic? Also, how will things like reST react to metadata in this format?